### PR TITLE
add codesnippet samples and javadocs

### PR DESCRIPTION
**Pull request by XiaofeiCao**
address https://github.com/Azure/azure-sdk-for-java/issues/34281

codesnippets are in `AzureResourceManager`. 
On `AzureResourceManager`, there's initialization codesnippet for itself.
Other codesnippets are on its methods. 
1. For `authenticate()` and `configure()`, add initialization codesnippet.
2. For `virtualMachines()`, `storageAccounts()`, ..., etc, add convenience layer sample codes.

First step, added samples from README.MD. 

Next step would be add samples for other popular RPs.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes;
+import com.azure.resourcemanager.resources.fluentcore.model.Creatable;
+import com.azure.resourcemanager.storage.models.PublicAccess;
+import com.azure.resourcemanager.storage.models.StorageAccount;
+import com.azure.resourcemanager.storage.models.StorageAccountSkuType;
+
+public class AzureResourceManagerJavaDocSamples {
+    public void authenticateUsingTokenCredentialAndAzureProfile() {
+        // BEGIN: com.azure.resourcemanager.azureResourceManager.authenticate#credential-profile
+        AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java:32]: Recommend to use the overload of specifying subscriptionId

File: sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    public void restartVirtualMachineAsync(AzureResourceManager azure,
+                                           String resourceGroupName) {
+        // BEGIN: com.azure.resourcemanager.azureResourceManager.virtualMachines.restartVirtualMachineAsync
+        azure.virtualMachines().listByResourceGroupAsync(resourceGroupName)
+            .flatMap(VirtualMachine::restartAsync)
+            //...
+            // END: com.azure.resourcemanager.azureResourceManager.virtualMachines.restartVirtualMachineAsync
+            .blockLast();
+    }
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java:121]: OK, I see. For batch restart, it maybe better for async. But for other create, seems sync is enough.

File: sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        VirtualMachine windowsVM = azure.virtualMachines()
+            .define(windowsVMName)
+            .withRegion(region)
+            .withNewResourceGroup(resourceGroupName)
+            .withNewPrimaryNetwork("10.0.0.0/28")
+            .withPrimaryPrivateIPAddressDynamic()
+            .withoutPrimaryPublicIPAddress()
+            .withPopularWindowsImage(KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2012_R2_DATACENTER)
+            .withAdminUsername(userName)
+            .withAdminPassword(password)
```
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java:77]: We don't recommend using passwords?

[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java:77]: At least not for Linux. It was blocked by policy in our tenant.

Windows likely still be fine (as seems the only way?), Personally I'd use Linux with ssh.

[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java]: Got it. Changed to linux with ssh.

File: sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/AzureResourceManagerJavaDocSamples.java]: Changed. Thanks!

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**